src/v/cluster/tests/topic_configuration_compat_test.cc

```diff
@@ -71,6 +74,36 @@ struct topic_configuration_assignment {
     std::vector<cluster::partition_assignment> assignments;
 };
 
+struct incremental_topic_updates {
```

Comment:
That should be `incremental_topic_updates_v2`. Also it would be nice to test v1 <-> v3 interop.